src/main/java/eu/dissco/backend/service/SpecimenService.java

```diff
@@ -167,42 +187,76 @@ private JsonNode generateIdNode() {
     return node;
   }
 
-  private DigitalSpecimen mapResultToSpecimen(JsonNode result) throws UnprocessableEntityException {
+  private DigitalSpecimen mapResultToSpecimen(JsonNode result) {
     var digitalSpecimen = result.get("digitalSpecimen");
     var attributes = digitalSpecimen.get("ods:attributes");
-    DigitalSpecimen ds;
-
-    try {
-      ds =  new DigitalSpecimen(
-          result.get("id").asText(),
-          result.get("midsLevel").asInt(),
-          result.get("version").asInt(),
-          Instant.ofEpochSecond(result.get("created").asInt()),
-          digitalSpecimen.get("ods:type").asText(),
-          digitalSpecimen.get("ods:physicalSpecimenId").asText(),
-          attributes.get("ods:physicalSpecimenIdType").asText(),
-          attributes.get("ods:specimenName").asText(),
-          attributes.get(ORGANISATION_ID).asText(),
-          attributes.get("ods:datasetId").asText(),
-          attributes.get("ods:physicalSpecimenCollection").asText(),
-          attributes.get("ods:sourceSystemId").asText(),
-          attributes,
-          digitalSpecimen.get("ods:originalAttributes"),
-          digitalSpecimen.get("ods:attributes").get("dwca:id").asText()
-      );
-    } catch (NullPointerException npe){
-      throw new UnprocessableEntityException("Unable to map given specimen and version to DigitalSpecimenObject");
-    }
-    return ds;
+    return new DigitalSpecimen(
+        result.get("id").asText(),
+        result.get(MIDS_LEVEL).asInt(),
+        result.get("version").asInt(),
+        Instant.ofEpochSecond(result.get("created").asInt()),
+        digitalSpecimen.get("ods:type").asText(),
+        digitalSpecimen.get("ods:physicalSpecimenId").asText(),
+        attributes.get("ods:physicalSpecimenIdType").asText(),
+        attributes.get("ods:specimenName").asText(),
+        attributes.get(ORGANISATION_ID).asText(),
+        attributes.get("ods:datasetId").asText(),
+        attributes.get("ods:physicalSpecimenCollection").asText(),
+        attributes.get("ods:sourceSystemId").asText(),
+        attributes,
+        digitalSpecimen.get("ods:originalAttributes"),
+        digitalSpecimen.get("ods:attributes").get("dwca:id").asText()
+    );
   }
 
-  private JsonApiListResponseWrapper wrapListResponse(List<DigitalSpecimen> digitalSpecimenList, int pageSize, int pageNumber, String path){
-    List<JsonApiData> dataNodePlusOne = new ArrayList<>();
-    digitalSpecimenList.forEach(specimen -> dataNodePlusOne.add(new JsonApiData(specimen.id(), specimen.type(), specimen, mapper)));
+  private JsonApiListResponseWrapper wrapListResponse(List<DigitalSpecimen> digitalSpecimenList,
+      int pageSize, int pageNumber, String path) {
+    var dataNodePlusOne = digitalSpecimenList.stream().map(specimen -> new JsonApiData(specimen.id(), specimen.type(), specimen, mapper)).toList();
```

Comment:
Why change `.forEach(...)` to `.stream.map(...)` ? Just curious if there's performance differences

Reply:
There is nothing wrong with the forEach but I don't really like that we init a list and then loop over another to fill it. This is bit more concise and just returns a new list of the values we want. I often try to make methods to return the object instead of munipulate a parameter. So instead of `void something(List<X> x)` and in the method add things I think it is more readable to have `List<X> something()`.